In [1]:
import pandas as pd

- dataframe : table
- series : column

In [48]:
#numpy와 비교
import numpy as np
data = np.array([1, 2, 3, 4, 5])
data

array([1, 2, 3, 4, 5])

In [15]:
data = pd.Series([1, 2, 3, 4, 5])
data

0    1
1    2
2    3
3    4
4    5
dtype: int64

In [16]:
data_2 = np.array([[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]])
data_2

array([[ 1,  2,  3,  4,  5],
       [ 6,  7,  8,  9, 10]])

In [17]:
data_2 = pd.Series([[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]])
data_2

0     [1, 2, 3, 4, 5]
1    [6, 7, 8, 9, 10]
dtype: object

In [18]:
data_2 = pd.DataFrame([[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]],
                      columns=['a', 'b', 'c', 'd', 'e'],
                     index=['one', 'two'])
data_2

a  b  c  d   e
one  1  2  3  4   5
two  6  7  8  9  10

In [13]:
data_2['a']

one    1
two    6
Name: a, dtype: int64

In [19]:
data_2.loc['one'] # = iloc = ix

a    1
b    2
c    3
d    4
e    5
Name: one, dtype: int64

In [20]:
data_2.iloc[0:1] # iloc 은 순서로 접근한다!

a  b  c  d  e
one  1  2  3  4  5

In [24]:
data_2.ix[0:1]

C:\ProgramData\Anaconda3\envs\finance\lib\site-packages\ipykernel_launcher.py:1: DeprecationWarning: 
.ix is deprecated. Please use
.loc for label based indexing or
.iloc for positional indexing

See the documentation here:
http://pandas.pydata.org/pandas-docs/stable/indexing.html#ix-indexer-is-deprecated
  """Entry point for launching an IPython kernel.


a  b  c  d  e
one  1  2  3  4  5

In [27]:
data_3 = pd.DataFrame([[1,2,3],[4,5,6]])
data_3

0  1  2
0  1  2  3
1  4  5  6

In [28]:
data_3.iloc[0]

0    1
1    2
2    3
Name: 0, dtype: int64

In [29]:
data_3.iloc[1]

0    4
1    5
2    6
Name: 1, dtype: int64

In [30]:
data_3.loc[1]

0    4
1    5
2    6
Name: 1, dtype: int64

In [31]:
data_3.ix[0]

C:\ProgramData\Anaconda3\envs\finance\lib\site-packages\ipykernel_launcher.py:1: DeprecationWarning: 
.ix is deprecated. Please use
.loc for label based indexing or
.iloc for positional indexing

See the documentation here:
http://pandas.pydata.org/pandas-docs/stable/indexing.html#ix-indexer-is-deprecated
  """Entry point for launching an IPython kernel.


0    1
1    2
2    3
Name: 0, dtype: int64

In [35]:
data_3.iloc[0:2]

0  1  2
0  1  2  3
1  4  5  6

In [41]:
data_3.loc[0:1]

0  1  2
0  1  2  3
1  4  5  6

In [167]:
data = np.array([i for i in range(100)]).reshape(10,10)
data_3=pd.DataFrame(data)
data_3

0   1   2   3   4   5   6   7   8   9
0   0   1   2   3   4   5   6   7   8   9
1  10  11  12  13  14  15  16  17  18  19
2  20  21  22  23  24  25  26  27  28  29
3  30  31  32  33  34  35  36  37  38  39
4  40  41  42  43  44  45  46  47  48  49
5  50  51  52  53  54  55  56  57  58  59
6  60  61  62  63  64  65  66  67  68  69
7  70  71  72  73  74  75  76  77  78  79
8  80  81  82  83  84  85  86  87  88  89
9  90  91  92  93  94  95  96  97  98  99

In [54]:
data_3.iloc[1:5]

0   1   2   3   4   5   6   7   8   9
1  10  11  12  13  14  15  16  17  18  19
2  20  21  22  23  24  25  26  27  28  29
3  30  31  32  33  34  35  36  37  38  39
4  40  41  42  43  44  45  46  47  48  49

In [55]:
data_3.loc[1:5]

0   1   2   3   4   5   6   7   8   9
1  10  11  12  13  14  15  16  17  18  19
2  20  21  22  23  24  25  26  27  28  29
3  30  31  32  33  34  35  36  37  38  39
4  40  41  42  43  44  45  46  47  48  49
5  50  51  52  53  54  55  56  57  58  59

In [56]:
data_3.ix[1:5]

C:\ProgramData\Anaconda3\envs\finance\lib\site-packages\ipykernel_launcher.py:1: DeprecationWarning: 
.ix is deprecated. Please use
.loc for label based indexing or
.iloc for positional indexing

See the documentation here:
http://pandas.pydata.org/pandas-docs/stable/indexing.html#ix-indexer-is-deprecated
  """Entry point for launching an IPython kernel.


0   1   2   3   4   5   6   7   8   9
1  10  11  12  13  14  15  16  17  18  19
2  20  21  22  23  24  25  26  27  28  29
3  30  31  32  33  34  35  36  37  38  39
4  40  41  42  43  44  45  46  47  48  49
5  50  51  52  53  54  55  56  57  58  59

In [175]:
data_3.loc[5][4]

54

In [174]:
data_3[4][5]

54

In [184]:
data_3[3:5][3]

3    33
4    43
Name: 3, dtype: int32

In [188]:
data_3[3].loc[3:4]

3    33
4    43
Name: 3, dtype: int32

In [96]:
data_3[3].iloc[3:5]

3    33
4    43
Name: 3, dtype: int32

In [189]:
data_3[3:5][3]

3    33
4    43
Name: 3, dtype: int32

In [190]:
data_3[3][3:5]

3    33
4    43
Name: 3, dtype: int32

In [128]:
data_3.tail(3)

0   1   2   3   4   5   6   7   8   9
7  70  71  72  73  74  75  76  77  78  79
8  80  81  82  83  84  85  86  87  88  89
9  90  91  92  93  94  95  96  97  98  99

In [126]:
data_3.shape

(10, 10)

In [131]:
data_3.columns.dtype

dtype('int64')

In [194]:
list(data_3)

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

In [200]:
cols = list(data_3)
cols.insert(0,cols.pop(9))

In [201]:
cols

[9, 0, 1, 2, 3, 4, 5, 6, 7, 8]

In [202]:
data_3[cols]

9   0   1   2   3   4   5   6   7   8
0   9   0   1   2   3   4   5   6   7   8
1  19  10  11  12  13  14  15  16  17  18
2  29  20  21  22  23  24  25  26  27  28
3  39  30  31  32  33  34  35  36  37  38
4  49  40  41  42  43  44  45  46  47  48
5  59  50  51  52  53  54  55  56  57  58
6  69  60  61  62  63  64  65  66  67  68
7  79  70  71  72  73  74  75  76  77  78
8  89  80  81  82  83  84  85  86  87  88
9  99  90  91  92  93  94  95  96  97  98

In [203]:
cols[1]=100

In [206]:
cols

[9, 100, 1, 2, 3, 4, 5, 6, 7, 8]

In [207]:
data_3[cols]

KeyError: '[100] not in index'

In [209]:
data_3

0   1   2   3   4   5   6   7   8   9
0   0   1   2   3   4   5   6   7   8   9
1  10  11  12  13  14  15  16  17  18  19
2  20  21  22  23  24  25  26  27  28  29
3  30  31  32  33  34  35  36  37  38  39
4  40  41  42  43  44  45  46  47  48  49
5  50  51  52  53  54  55  56  57  58  59
6  60  61  62  63  64  65  66  67  68  69
7  70  71  72  73  74  75  76  77  78  79
8  80  81  82  83  84  85  86  87  88  89
9  90  91  92  93  94  95  96  97  98  99

In [210]:
data_3[0]=data_3[9]
data_3

0   1   2   3   4   5   6   7   8   9
0   9   1   2   3   4   5   6   7   8   9
1  19  11  12  13  14  15  16  17  18  19
2  29  21  22  23  24  25  26  27  28  29
3  39  31  32  33  34  35  36  37  38  39
4  49  41  42  43  44  45  46  47  48  49
5  59  51  52  53  54  55  56  57  58  59
6  69  61  62  63  64  65  66  67  68  69
7  79  71  72  73  74  75  76  77  78  79
8  89  81  82  83  84  85  86  87  88  89
9  99  91  92  93  94  95  96  97  98  99

In [219]:
dic = {'a':1,'b':2}
dic['a'] = 3
dic['c'] = 5
dic

{'a': 3, 'b': 2, 'c': 5}

In [225]:
df_1 = pd.DataFrame([[1,2,3],[4,5,6]],
                    columns =['a','b','c'])
df_1

a  b  c
0  1  2  3
1  4  5  6

In [227]:
df_1['a'] = df_1['b']
df_1['d'] = df_1['c']
df_1

a  b  c  d
0  2  2  3  3
1  5  5  6  6

In [228]:
df_1['a'] +=3
df_1

a  b  c  d
0  5  2  3  3
1  8  5  6  6

In [229]:
df_1['a']=df_1['b'] 

In [233]:
df_1.a = df_1.b *df_1.c
df_1

a  b  c  d
0   6  2  3  3
1  30  5  6  6

In [268]:
df_2 = pd.DataFrame([['1,000','2,000','3,000'],
                    ['1','2','3']])

In [269]:
df_2

0      1      2
0  1,000  2,000  3,000
1      1      2      3

In [270]:
df_2.stack()

0  0    1,000
   1    2,000
   2    3,000
1  0        1
   1        2
   2        3
dtype: object

In [274]:
df_2 = df_2.stack().str.replace(",","").unstack()

In [277]:
df_2.astype(int)

0     1     2
0  1000  2000  3000
1     1     2     3

In [284]:
multi_ind = pd.MultiIndex.from_tuples([('a',0), ('a',1)])

multi_col = [['b','b','b'],[0,1,2]]

In [287]:
df_mul = pd.DataFrame(df_2, columns=multi_col, index=multi_ind)
df_mul

b        
      0   1   2
a 0 NaN NaN NaN
  1 NaN NaN NaN

In [288]:
df_2

0     1     2
0  1000  2000  3000
1     1     2     3

### Titanic data

In [1]:
mkdir Dataset

In [6]:
dataset = pd.read_csv('Dataset/train.csv')
dataset

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
5              6         0       3   
6              7         0       1   
7              8         0       3   
8              9         1       3   
9             10         1       2   
10            11         1       3   
11            12         1       1   
12            13         0       3   
13            14         0       3   
14            15         0       3   
15            16         1       2   
16            17         0       3   
17            18         1       2   
18            19         0       3   
19            20         1       3   
20            21         0       2   
21            22         1       2   
22            23         1       3   
23            24         1       1   
24            25         0       3   
25            26         1       3   
26            27         0       3   
27            28         0       1   
28            29         1       3   
29            30         0       3   
..           ...       ...     ...   
861          862         0       2   
862          863         1       1   
863          864         0       3   
864          865         0       2   
865          866         1       2   
866          867         1       2   
867          868         0       1   
868          869         0       3   
869          870         1       3   
870          871         0       3   
871          872         1       1   
872          873         0       1   
873          874         0       3   
874          875         1       2   
875          876         1       3   
876          877         0       3   
877          878         0       3   
878          879         0       3   
879          880         1       1   
880          881         1       2   
881          882         0       3   
882          883         0       3   
883          884         0       2   
884          885         0       3   
885          886         0       3   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
5                                     Moran, Mr. James    male   NaN      0   
6                              McCarthy, Mr. Timothy J    male  54.0      0   
7                       Palsson, Master. Gosta Leonard    male   2.0      3   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
10                     Sandstrom, Miss. Marguerite Rut  female   4.0      1   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
12                      Saundercock, Mr. William Henry    male  20.0      0   
13                         Andersson, Mr. Anders Johan    male  39.0      1   
14                Vestrom, Miss. Hulda Amanda Adolfina  female  14.0      0   
15                    Hewlett, Mrs. (Mary D Kingcome)   female  55.0      0   
16                                Rice, Master. Eugene    male   2.0      4   
17                        Williams, Mr. Charles Eugene    male   NaN      0   
18   Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.0      1   
19                             Masselmani, Mrs. Fatima  female   NaN

In [7]:
dataset.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [8]:
dataset.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

In [9]:
dataset.loc[400:404]

PassengerId  Survived  Pclass                            Name     Sex  \
400          401         1       3              Niskanen, Mr. Juha    male   
401          402         0       3                 Adams, Mr. John    male   
402          403         0       3        Jussila, Miss. Mari Aina  female   
403          404         0       3  Hakkarainen, Mr. Pekka Pietari    male   
404          405         0       3         Oreskovic, Miss. Marija  female   

      Age  SibSp  Parch             Ticket     Fare Cabin Embarked  
400  39.0      0      0  STON/O 2. 3101289   7.9250   NaN        S  
401  26.0      0      0             341826   8.0500   NaN        S  
402  21.0      1      0               4137   9.8250   NaN        S  
403  28.0      1      0   STON/O2. 3101279  15.8500   NaN        S  
404  20.0      0      0             315096   8.6625   NaN        S

In [10]:
dataset.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [13]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Survived       891 non-null int64
Pclass         891 non-null int64
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Cabin          204 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 83.6+ KB


- Age는 714개 / 총 사람 수 891 ->결측치(누락 데이터)가 있다.

#### DataFrame 병합

In [92]:
df_1 = pd.DataFrame([['알라딘', 26],['자스민', 25]], columns=['name','age'])
df_1

name  age
0  알라딘   26
1  자스민   25

In [93]:
df_2 = pd.DataFrame([['지니', 3000],['자카',50]], columns=['name','age'])
df_2

name   age
0   지니  3000
1   자카    50

In [94]:
df = df_1.append(df_2, ignore_index=True)
df

name   age
0  알라딘    26
1  자스민    25
2   지니  3000
3   자카    50

In [95]:
df_3 = pd.DataFrame([['알라딘', '남자'],['자스민', '여자']], columns=['name','sex'])
df_3

name sex
0  알라딘  남자
1  자스민  여자

In [96]:
df = df_1.merge(df_3)
df

name  age sex
0  알라딘   26  남자
1  자스민   25  여자

In [97]:
df_4 = pd.DataFrame([['알라딘', '왕자'],['자스민', '공주']], columns=['name','job'])
df_4

name job
0  알라딘  왕자
1  자스민  공주

In [98]:
df = df.merge(df_4)
df

name  age sex job
0  알라딘   26  남자  왕자
1  자스민   25  여자  공주

In [99]:
df_5 = pd.DataFrame(['무직','공주'], columns=['job'])
df_5

job
0  무직
1  공주

In [100]:
pd.concat([df_1, df_2])

name   age
0  알라딘    26
1  자스민    25
0   지니  3000
1   자카    50

In [101]:
df_1.join(df_5)

name  age job
0  알라딘   26  무직
1  자스민   25  공주

In [102]:
df_genie = pd.DataFrame({'name': ['지니'], 'age' : [3000], 'sex' : ['남자'], 'job':['요정']})
df_genie            

name   age sex job
0   지니  3000  남자  요정

In [103]:
df = df.append(df_genie, ignore_index=True)
df

name   age sex job
0  알라딘    26  남자  왕자
1  자스민    25  여자  공주
2   지니  3000  남자  요정

In [110]:
df_country = pd.DataFrame({'name' : ['알라딘', '자스민', '자카'], '국적':['아라비아', '아라비아', '아라비아']})
df_country

name    국적
0  알라딘  아라비아
1  자스민  아라비아
2   자카  아라비아

In [125]:
data = df.merge(df_country, how = 'right') # how = left, right, inner, outer
data

#inner : 서로 같은 index만 뽑아서 merge하자. (how값의 default는 inner)
#outer : 전부 다 합치고 겹치는 값은 merge, 없는 값은 NaN으로 merge
#left : df.merge(df_country) / 왼쪽인 df에서 row(행)을 일단 다 쓰고 여기에 right를 합치자.
#rigth : 위와 반대로

name   age  sex  job    국적
0  알라딘  26.0   남자   왕자  아라비아
1  자스민  25.0   여자   공주  아라비아
2   자카   NaN  NaN  NaN  아라비아

In [131]:
df_left = pd.DataFrame({'key' : ['A', 'B', 'C'], 'left' : [1,2,3]})
df_left

key  left
0   A     1
1   B     2
2   C     3

In [132]:
df_right = pd.DataFrame({'key' : ['A', 'B', 'D'], 'right' : [4,5,6]})
df_right

key  right
0   A      4
1   B      5
2   D      6

In [134]:
test = df_left.merge(df_right, how='left')
test

key  left  right
0   A     1    4.0
1   B     2    5.0
2   C     3    NaN

In [135]:
test = df_left.merge(df_right, how='right')
test

key  left  right
0   A   1.0      4
1   B   2.0      5
2   D   NaN      6

In [136]:
test = df_left.merge(df_right, how='outer')
test

key  left  right
0   A   1.0    4.0
1   B   2.0    5.0
2   C   3.0    NaN
3   D   NaN    6.0

- merge(on) #기준열 설정

In [137]:
test = df_left.merge(df_right, how='right', on = '')
test

KeyError: ''

- left, right 모두에게 'data'라는 이름의 columns를 추가하되, left와 right에 다른 값을 넣어보세요

In [185]:
df_left2 = pd.DataFrame({'key' : ['A', 'B', 'C'], 'left' : [1,2,3], 'data' : ['100','200','300']})
df_right2 = pd.DataFrame({'key' : ['A', 'B', 'D'], 'right' : [4,5,6],'data' : ['a','b','c']})

In [190]:
df = df_left2.merge(df_right2, on='key')
df

key  left data_x  right data_y
0   A     1    100      4      a
1   B     2    200      5      b

- 반대로 기준열로 삼아야하는 컬럼이 서로 이름이 다른경우

In [199]:
df_left3 = pd.DataFrame({'key' : ['A', 'B', 'C'], 'left' : [1,2,3]})
df_right3 = pd.DataFrame({'name' : ['A', 'B', 'D'], 'right' : [4,5,6]})

In [203]:
df = df_left3.merge(df_right3, left_on='key', right_on='name', how ='outer')
df

key  left name  right
0    A   1.0    A    4.0
1    B   2.0    B    5.0
2    C   3.0  NaN    NaN
3  NaN   NaN    D    6.0

In [211]:
df = pd.concat([df_left3, df_right3]) #concat은 값을 아래로 쌓은다.
df

C:\ProgramData\Anaconda3\envs\finance\lib\site-packages\ipykernel_launcher.py:1: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  """Entry point for launching an IPython kernel.


key  left name  right
0    A   1.0  NaN    NaN
1    B   2.0  NaN    NaN
2    C   3.0  NaN    NaN
0  NaN   NaN    A    4.0
1  NaN   NaN    B    5.0
2  NaN   NaN    D    6.0

In [212]:
df = pd.concat([df_left3, df_right3], axis=1) #concat은 값을 아래로 쌓은다, '0'이 default. axis=1은 옆으로쌓인다.
df

key  left name  right
0   A     1    A      4
1   B     2    B      5
2   C     3    D      6

In [213]:
data = pd.DataFrame({'name':['알라딘','자스민','지니'],'age':[25,25,3000]})
data

name   age
0  알라딘    25
1  자스민    25
2   지니  3000

-data라는 dataframe에 'name'이 '자카'인 데이터가 있는지 궁금하다면?


In [226]:
[data['name']=='지니']

[0    False
 1    False
 2     True
 Name: name, dtype: bool]

In [225]:
data[data['name']=='지니']

name   age
2   지니  3000

In [223]:
data[data['name']=='자카']

Empty DataFrame
Columns: [name, age]
Index: []

- data에서 알라딘과 자스민만 뽑아보세요

In [241]:
data[data['name'].isin(['알라딘','자스민'])]

name  age
0  알라딘   25
1  자스민   25

- age 기준으로 정렬을 하고 싶다면?

In [243]:
data.sort_values(by=['age'], ascending=False)

name   age
2   지니  3000
0  알라딘    25
1  자스민    25

- csv로 저장하고 싶으면?

In [266]:
data = pd.DataFrame({'name':['알라딘','자스민','지니'],'age':[25,25,3000]})
data

name   age
0  알라딘    25
1  자스민    25
2   지니  3000

In [260]:
data.to_csv('aladin.csv',index=False) #보통 index = False

In [267]:
data = pd.read_csv('aladin.csv')
data

name   age
0  알라딘    25
1  자스민    25
2   지니  3000

In [268]:
data.to_excel('aladin.xls', index=False)

In [269]:
data = pd.read_excel('aladin.xls')
data

name   age
0  알라딘    25
1  자스민    25
2   지니  3000

- name 칼럼을 Index로 사용하고 싶다면?

In [272]:
data = data.set_index('name')
data

age
name      
알라딘     25
자스민     25
지니    3000

In [279]:
#저장된 csv파일을 읽어들일 때
data= pd.read_csv('aladin.csv', index_col='name')
data

age
name      
알라딘     25
자스민     25
지니    3000